analyzers/src/SonarAnalyzer.CSharp/Rules/LossOfFractionInDivision.cs

```diff
@@ -43,8 +47,7 @@ public sealed class LossOfFractionInDivision : SonarDiagnosticAnalyzer
                 {
                     var division = (BinaryExpressionSyntax)c.Node;
 
-                    if (!(c.SemanticModel.GetSymbolInfo(division).Symbol is IMethodSymbol symbol)
-                        || symbol.ContainingType == null
```

Comment:
I'd leave defensive coding iffor the case of the IDE (if it somehow doesn't manage to compute smth?)